azurerm/internal/services/dns/dns_zone_resource.go

```diff
@@ -253,6 +259,12 @@ func resourceDnsZoneRead(d *schema.ResourceData, meta interface{}) error {
 		return fmt.Errorf("setting `soa_record`: %+v", err)
 	}
 
+	fqdn := ""
+	if rsResp.Fqdn != nil {
+		fqdn = *rsResp.Fqdn
+	}
+	d.Set("fqdn", fqdn)
```

Comment:
Thoughts on shortening lines 262-266 to just. d.Set does a pretty good job of handling nil base types as long as we don't point to it. Lists and Sets are the ones we have to make sure we always set to at least an empty value

```suggestion
	d.Set("fqdn", rsResp.Fqdn)
```